presto-main/src/main/java/com/facebook/presto/operator/scalar/sql/ArraySqlFunctions.java

```diff
@@ -136,4 +136,22 @@ public static String removeNulls()
     {
         return "RETURN IF(none_match(input, x -> x is null), input, filter(input, x -> x is not null))";
     }
+
+    @SqlInvokedScalarFunction(value = "array_top_n", deterministic = true, calledOnNullInput = true)
+    @Description("Returns top N elements of a given array, using natural descending order.")
+    @TypeParameter("T")
+    @SqlParameters({@SqlParameter(name = "input", type = "array(T)"), @SqlParameter(name = "n", type = "int")})
+    @SqlType("array<T>")
+    public static String arrayTopN()
+    { return "RETURN IF(n < 0, fail('n must be greater than or equal to 0'), SLICE(ARRAY_SORT_DESC(input), 1, n))"; }
```

Comment:
Be more descriptive here. I know presto has this bad error messages all over but at least when we have a chance, lets' fix it. Say something like:

fail('Paramater n: '  || cast(n as varchar) || ' to ARRAY_TOP_N is negative')

Reply:
Thanks, that's a good point. Will update PR.